meerkat/mixins/mapping.py

```diff
@@ -87,19 +87,25 @@ def map(
                     batch,
                     range(start_index, end_index),
                     materialize=materialize,
+                    **kwargs,
                 )
 
                 # Pull out information
                 output = function_properties.output
                 dtype = function_properties.output_dtype
                 is_mapping = isinstance(output, Mapping)
+                is_type_mapping = isinstance(output_type, Mapping)
```

Comment:
We should probably raise a `ValueError` if the output is not mapping but the type is a mapping.